src/lib/operators/aggregate_hash.cpp

```diff
@@ -27,43 +27,94 @@
 namespace {
 using namespace opossum;  // NOLINT
 
-// Given an AggregateKey key, and a RowId row_id where this AggregateKey was encountered, this first checks if the
-// AggregateKey was seen before. If not, a new aggregate result is inserted into results and connected to the row id.
-// This is important so that we can reconstruct the original values later. In any case, a reference to the result is
-// returned so that result information, such as the aggregate's count or sum, can be modified by the caller.
-template <typename ResultIds, typename Results, typename AggregateKey>
-typename Results::reference get_or_add_result(ResultIds& result_ids, Results& results, const AggregateKey& key,
-                                              const RowID& row_id) {
-  // Get the result id for the current key or add it to the id map
+// For a given AggregateKey key, and a RowId row_id where this AggregateKey was encountered, this returns a reference
```

Comment:
> `get_or_add_result` is called once per row when iterating over a column that is to be aggregated. The row's `key` has been calculated as part of `_partition_by_groupby_keys`. We also pass in the `row_id` of that row. This row id is stored in `Results` so that we can later use it to reconstruct the values in the group-by columns.

Better?

Reply:
Jep. 👍